static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -166,7 +165,7 @@ function WidgetBuilder({
     location.query.defaultWidgetQuery
   );
 
-  const isEditing = defined(widgetIndex);
+  const isEditing = Number.isInteger(parseInt(widgetIndex, 10));
```

Comment:
do we need to do these extra checks? wouldn't it be fine if we would only use `Number()` ?

Reply:
if the user updates the URL for instance to be `10.50`, this would still pass, although it's not a valid index

Reply:
Good point, I've updated the component to have `isValidWidgetIndex` which checks the bounds as well as whether the value is an integer, there's also a test for it now